lib/constructs/AgentDefinitionBuilder.ts

```diff
@@ -209,6 +209,35 @@ export class AgentDefinitionBuilder extends Construct {
         return this;
     }
 
+    /**
+     * To allow agent to generate, run, and troubleshoot your application code in a secure test environment.
+     * We need to add an action group with the parentActionGroupSignature 
+     * field set to AMAZON.CodeInterpreter.
+     * @returns AgentDefinitionBuilder object with a new action set.
+     */
+    public withCodeInterpreter() {
+        const codeInterpreterAction: bedrock.CfnAgent.AgentActionGroupProperty = {
+            actionGroupName: CODE_INTERPRETER_ACTION_NAME,
+            actionGroupState: "ENABLED",
+            parentActionGroupSignature: CODE_INTERPRETER_PARENT_SIGNATURE,
+        };
+
+        // Append the codeInterpreter AG to the agent definition.
+        this.agentDefinition = {
+            ...this.agentDefinition,
+            actionGroups: [codeInterpreterAction]
```

Comment:
This will replace the actionGroups. Suggest to update this to be additive.
```
const existingActions = <[]>this.agentDefinition?.actionGroups || [];

        // Append the agentActionGroups to the agent definition.
        this.agentDefinition = {
            ...this.agentDefinition,
            actionGroups: [
                ...existingActions,
                ...codeInterpreterAction
            ],
        };
```